include/bluetooth/uuid.h

```diff
@@ -480,7 +483,7 @@ void bt_uuid_to_str(const struct bt_uuid *uuid, char *str, size_t len);
  *
  *  @return String representation of the UUID given
  */
-const char *bt_uuid_str(const struct bt_uuid *uuid);
+#define bt_uuid_str(_uuid) log_strdup(bt_uuid_str_real(_uuid))
```

Comment:
Is this a safe thing to do? It seems to encourage memory leaks, since the log processor is the only thing that frees `log_strdup()`-ed strings. I would have expected changing the `LOG_XYZ()` call sites to use `log_strdup(bt_uuid_str())` instead, or a name that makes it really obvious what's going on, like `bt_uuid_log_strdup()`.

@nordic-krch though I wonder if there's some way to mark that a string is safely not going to change, even if it's not in ROM? Just a thought here...

Reply:
@mbolivar that's a fair question. I followed the same style as has been used for our other string conversion helpers:

https://github.com/zephyrproject-rtos/zephyr/blob/d3977996088a356107257adb828be649ec2fa7e3/subsys/bluetooth/common/log.h#L52-L58

You're right that if anyone tried to use these for anything else than logging we'd end up with memory leaks.

Reply:
> though I wonder if there's some way to mark that a string is safely not going to change, even if it's not in ROM? Just a thought here...

The only thing that comes to my mind is to use custom format specifier for such thing (e.g. %zs aka zephyr string) but then there is this `printf_like` check. Is it possible to make it aware of custom format specifiers?

Reply:
Anyway we were considering extending format specifiers with things like ble address so it would be good to know if `printf_like` check can be extended or configured to not generate warning when format specifier is unknown.

Reply:
@nordic-krch the only example I know of extending the format specifiers is what Linux has done:
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/core-api/printk-formats.rst

The notable thing there is that these are all extensions of %p, i.e. to printf_like these will all look like pointers (since that's what they are - it's just the rendering of data behind the pointer that's custom) and won't therefore cause warnings. If we'd follow this then we can do slightly better than the current log_strdup() by introducing a log_memdup(). The amount of memory needing duplication should roughly drop in half, and we'd also move move some CPU usage to the logging thread since that's where the "binary to hex" or "address to string" conversion would happen.

Reply:
> The only thing that comes to my mind is to use custom format specifier for such thing (e.g. %zs aka zephyr string) but then there is this printf_like check. 

I was thinking a special linker section which we could allocate "initialized-once" strings into. Or would that add too much overhead to this check?